### PR TITLE
rpmsgsocket：fix build warning

### PR DESCRIPTION
**Pull request by yintao707**
## Summary
1.  fix build warning, add header files
2. fix redefine warning

## Impact
none

## Testing
sim/ vela ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.** 

Here's why and how to improve it:

* **Insufficient Summary:**
    * **Missing "Why":**  You haven't explained *why* the build warnings and redefine issues exist in the first place. What's the underlying problem you're solving?
    * **Lack of Detail:**  "Fix build warning, add header files" and "fix redefine warning" are too vague.  Be specific about:
        * Which header files are added and why?
        * What was being redefined and how did you resolve it?
    * **Issue References:** If these problems are documented in NuttX issues, provide links.

* **Incomplete Impact Assessment:**
    * While "none" might be the correct answer for most sections, you should explicitly state "NO" for each to show you've considered them.
    * **Consider Potential Impacts:** Even seemingly small changes can have ripple effects. For example:
        * **Compatibility:** Did fixing the redefine change any function signatures or behavior that other code might rely on?
        * **Documentation:** If the added header files introduce new functions or data structures, documentation might be needed.

* **Inadequate Testing:**
    * **Insufficient Detail:** "sim/ vela ci" doesn't provide enough context. Specify:
        *  The exact simulator used (e.g., QEMU, Renode).
        *  The full Vela configuration or a link to your CI run.
    * **Missing Logs:** The "Testing logs before change" and "Testing logs after change" sections are empty. Include relevant log snippets that demonstrate:
        * The presence of the warnings/errors before your changes.
        * The absence of these issues after your changes.

**How to Improve Your PR:**

1. **Expand the Summary:** Provide a clear, concise explanation of the problem and your solution.
2. **Thoroughly Address Impact:**  Go through each impact category and either confirm "NO" or provide a detailed explanation if "YES."
3. **Provide Specific Testing Details:** Include the simulator, configuration, and relevant log snippets.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by cederom — APPROVED** [review]
Thank you @yintao707 :-)

File: examples/rpmsgsocket/rpsock_client.c [hunk 1]
```diff
@@ -62,7 +65,7 @@ struct rpsock_arg_s
  * Public Functions
  ****************************************************************************/
 
-static void *rpsock_send_thread(pthread_addr_t pvarg)
+static void *rpsock_send_thread(FAR void *pvarg)
```
Comment (pkarashchenko) [inline examples/rpmsgsocket/rpsock_client.c:68]:
```suggestion
static FAR void *rpsock_send_thread(FAR void *pvarg)
```

File: examples/rpmsgsocket/rpsock_client.c [hunk 2]
```diff
@@ -193,17 +195,18 @@ static int rpsock_unsync_test(struct rpsock_arg_s *args)
 
           if (args->check && ret > 4)
             {
-              intp   = (uint32_t *)(args->inbuf + (ALIGN_UP(total) - total));
-              checks = ret - (ALIGN_UP(total) - total);
+              checks = ret - (ALIGN_UP(total, 4) - total);
+              intp   = (uint32_t *)(args->inbuf +
```
Comment (pkarashchenko) [inline examples/rpmsgsocket/rpsock_client.c:199]:
```suggestion
              intp   = (FAR uint32_t *)(args->inbuf +
```

File: examples/rpmsgsocket/rpsock_server.c [hunk 3]
```diff
@@ -46,10 +49,10 @@ struct rpsock_arg_s
 };
 
 /****************************************************************************
- * Public Functions
+ * Private Functions
  ****************************************************************************/
 
-static void *rpsock_thread(pthread_addr_t pvarg)
+static void *rpsock_thread(FAR void *pvarg)
```
Comment (pkarashchenko) [inline examples/rpmsgsocket/rpsock_server.c:55]:
```suggestion
static FAR void *rpsock_thread(FAR void *pvarg)
```

File: examples/rpmsgsocket/rpsock_server.c [hunk 4]
```diff
@@ -46,10 +49,10 @@ struct rpsock_arg_s
...
 {
   struct rpsock_arg_s *args = pvarg;
```
Comment (pkarashchenko) [inline examples/rpmsgsocket/rpsock_server.c:57]:
```suggestion
  FAR struct rpsock_arg_s *args = pvarg;
```

File: examples/rpmsgsocket/rpsock_server.c [hunk 5]
```diff
@@ -266,7 +268,7 @@ static int rpsock_stream_server(int argc, char *argv[])
   return -errno;
 }
 
-static int rpsock_dgram_server(int argc, char *argv[])
+static int rpsock_dgram_server(int argc, FAR char *argv[])
 {
   struct sockaddr_rpmsg myaddr;
   struct rpsock_arg_s *args;
```
Comment (pkarashchenko) [inline examples/rpmsgsocket/rpsock_server.c:274]:
```suggestion
  FAR struct rpsock_arg_s *args;
```

File: examples/rpmsgsocket/rpsock_client.c (hunk 1 above)
Comment (yintao707) [reply examples/rpmsgsocket/rpsock_client.c]: done，thanks

**Review by xiaoxiang781216 — APPROVED** [review]

File: examples/rpmsgsocket/rpsock_client.c [hunk 6]
```diff
@@ -225,14 +228,14 @@ static int rpsock_unsync_test(struct rpsock_arg_s *args)
   return 0;
 }
 
-static int rpsock_stream_client(int argc, char *argv[])
+static int rpsock_stream_client(FAR char *argv[])
```
Comment (cederom) [inline examples/rpmsgsocket/rpsock_client.c:231]: no need for `argc`? just an example code?

File: examples/rpmsgsocket/rpsock_client.c [hunk 7]
```diff
@@ -225,14 +228,14 @@ static int rpsock_unsync_test(struct rpsock_arg_s *args)
...
   struct sockaddr_rpmsg myaddr;
   struct rpsock_arg_s args;
   bool nonblock = false;
   int cnt = 0;
-  char *outbuf;
-  char *inbuf;
+  FAR char *outbuf;
+  FAR char *inbuf;
   int sockfd;
   int ret;
```
Comment (cederom) [inline examples/rpmsgsocket/rpsock_client.c:241]: What happens in 
```
if (strcmp(argv[2], "nonblock") == 0)
```
when argv[2] is out of bounds? i.e. `argv<2`? will never happen?

Comment (cederom) [reply examples/rpmsgsocket/rpsock_client.c:241]: I know it was here before but maybe worth looking into in a free moment? :-)

Comment (yintao707) [reply examples/rpmsgsocket/rpsock_client.c:241]: int main(int argc, char *argv[])
{
  if (argc < 4)
    {
      printf("Usage: rpsock_client stream/dgram"
             " block/nonblock rp_name rp_cpu\n");
      return -EINVAL;
    }

if argc < 4, it will not call rpsock_stream_client, So I don't think it needs to be judged again

File: examples/rpmsgsocket/rpsock_client.c (hunk 6 above)
Comment (yintao707) [reply examples/rpmsgsocket/rpsock_client.c:231]: Thank you, I think it has already been determined in the main and should no longer be needed here
